components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1437,7 +1438,60 @@ public static Time getTime(Double value, String unit) {
     }
     return new Time(value, UNITS.SECOND);
   }
-  
+
+
+  /**
+   * Formats the input value for the position into a length of the given unit.
+   *
+   * @param value  the value of the position
+   * @param unit   the unit of the position. If {@code null} or invalid,
+   *               default to reference frame.
+   *
+   * @return       the position formatted as a {@link Length}
+   */
+  public static Length getPosition(Double value, Unit<Length> unit) {
+    if (value == null || value.isInfinite()) {
+      LOGGER.debug("Expected float value for position; got {}", value);
+      return null;
+    }
+
+    if (unit == null) unit = UNITS.REFERENCEFRAME;
```

Comment:
Explicitly passing a valid unit type is certainly what we have enforced in our readers so far.
Thus the proposal of handling  `null`/invalid units in the same way we handle `null`/invalid values is certainly sensible to me. I will adjust the method/tests accordingly.

What are the implications for default methods like `getPosition(Double)`? Is there a need to define such unit-less methods (and use the default unit specified in the model) for other consumers of the API than our own readers?